### PR TITLE
Don't register signal in thread

### LINKED ISSUE #10336: v1.5.0 breaks wandb hyperparameter sweeps in Colab
## 🐛 Bug

After upgrading to v1.5.0, wandb hyperparameter sweeps performed in Colab notebooks fail with one `UserWarning` and one `ValueError` raised:

```python
UserWarning: There is a wandb run already in progress and newly created instances of `WandbLogger` will reuse this run. If this is not desired, call `wandb.finish()` before instantiating `WandbLogger`.
```
```python
ValueError('signal only works in main thread')
```
`wandb` sweep terminates without performing training runs. Downgrading to `pl` v1.4.9 resolves the issue.

### To Reproduce

Reproduced at the end of this minimally modified BoringModel Colab notebook:
[https://colab.research.google.com/drive/16nylTt8jGAbiSfq7zr7YlLOohfIbFqHq?usp=sharing](https://colab.research.google.com/drive/16nylTt8jGAbiSfq7zr7YlLOohfIbFqHq?usp=sharing)

### Expected behavior

`wandb` sweep should run training loops without terminating due to these errors.

### Environment

* CUDA:
	- GPU:
		- Tesla V100-SXM2-16GB
	- available:         True
	- version:           11.1
* Packages:
	- numpy:             1.19.5
	- pyTorch_debug:     False
	- pyTorch_version:   1.9.0+cu111
	- pytorch-lightning: 1.5.0
	- tqdm:              4.62.3
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.7.12
	- version:           #1 SMP Sat Jun 5 09:50:34 PDT 2021
You can also fill out the list below manually.
-->

- PyTorch Lightning Version (e.g., 1.3.0):
- PyTorch Version (e.g., 1.8)
- Python version:
- OS (e.g., Linux):
- CUDA/cuDNN version:
- GPU models and configuration:
- How you installed PyTorch (`conda`, `pip`, source):
- If compiling from source, the output of `torch.__config__.show()`:
- Any other relevant information:

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

Fixes #10336

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

File: pytorch_lightning/trainer/connectors/signal_connector.py [hunk 1]
```diff
@@ -107,3 +107,11 @@ def _has_already_handler(self, signum: Signals) -> bool:
...
         except AttributeError:
             return False
+
+    @staticmethod
+    def _register_signal(signum: Signals, handlers: HandlersCompose) -> None:
+        try:
+            signal.signal(signum, handlers)
+        except ValueError as e:
+            if "signal only works in main thread" not in str(e):
+                raise e
```
Comment (kaushikb11) [inline pytorch_lightning/trainer/connectors/signal_connector.py:117]: Could we use https://docs.python.org/3.4/library/threading.html#threading.main_thread here?

**Review by awaelchli — APPROVED** [review]
other than @kaushikb11 comment LGTM

File: tests/trainer/connectors/test_signal_connector.py [hunk 2]
```diff
@@ -57,3 +58,16 @@ def training_step(self, batch, batch_idx):
...
+
+
+def _registering_signals():
+    with mock.patch.dict(os.environ, {"PL_FAULT_TOLERANT_TRAINING": "1"}):
+        trainer = Trainer()
+        trainer.signal_connector.register_signal_handlers()
+
+
+@RunIf(skip_windows=True)
+def test_signal_connector_in_thread():
```
Comment (carmocca) [inline tests/trainer/connectors/test_signal_connector.py:70]: Shouldn't this set the fault tolerant env variable?

File: pytorch_lightning/trainer/connectors/signal_connector.py (hunk 1 above)
Comment (tchaton) [reply pytorch_lightning/trainer/connectors/signal_connector.py:117]: Done !

File: tests/trainer/connectors/test_signal_connector.py (hunk 2 above)
Comment (tchaton) [reply tests/trainer/connectors/test_signal_connector.py:70]: Done !

**Review by kaushikb11 — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Review by carmocca — APPROVED** [review]
